### PR TITLE
[deprecated]Add New Catalog IDL

### PR DESCRIPTION
**Pull request by chanadian**
This is the IDL for the new DataCatalog service. It is serviced through this name `flyteidl.datacatalog.DataCatalogService`. 

Summary of service:
-------------------
It is a service where you can store artifacts and tag them with a string. We will use this for caching flyte executions as artifacts.

- Dataset: represents the project/domain/version/name of the collection of artifacts to store
- Artifact: represents an entity with a list of artifactData 
- ArtifactData: represents a literal value to store that belongs to the artifact
- Tag: A tag is a way to tag an artifact by name

Included in this PR are:
------------------------
- IDL protos under `/datacatalog` folder and generated `go` files
- DataCatalogServiceClient: a client that can talk to the service via grpc
- client_configs: connection configs for the above client

### CODE REVIEW

**Comment by chanadian** [comment]
👀 @jonathanburns @kumare3

File: protos/flyteidl/datacatalog/datacatalog.proto [hunk 1]
```diff
@@ -1,128 +1,93 @@
...
-package pb.lyft.datacatalog;
+package flyteidl.datacatalog;
+option go_package = "github.com/lyft/flyteidl/gen/pb-go/flyteidl/datacatalog";
 
-option go_package = "datacatalog";
-option py_generic_services = true;
+import "flyteidl/core/literals.proto";
 
-message Parameter {
+service DataCatalogService {
```
Comment (kumare3) [inline protos/flyteidl/datacatalog/datacatalog.proto]: documentation please

File: protos/flyteidl/datacatalog/datacatalog.proto [hunk 2]
```diff
@@ -1,128 +1,93 @@
...
-    repeated Parameter inputs = 3;
+message CreateDatasetResponse {
+
 }
 
-message GetRequest {
-    oneof id {
-        string provenance = 1;
-        ArtifactId artifact_id = 2;
+message GetDatasetRequest {
```
Comment (kumare3) [inline protos/flyteidl/datacatalog/datacatalog.proto]: docs

**Comment by kumare3** [comment]
2 things, missing docs and add a title.rst (look at other for example)

File: protos/flyteidl/datacatalog/datacatalog.proto (hunk 1 above)
Comment (chanadian) [reply protos/flyteidl/datacatalog/datacatalog.proto]: done.

File: protos/flyteidl/datacatalog/datacatalog.proto (hunk 2 above)
Comment (chanadian) [reply protos/flyteidl/datacatalog/datacatalog.proto]: done

**Comment by kumare3** [comment]
LGTM

**Comment by kumare3** [comment]
cc @tnsetting

**Pull request closed without merging**